src/numbers/xdr_large_int.js

```diff
@@ -238,16 +226,29 @@ export class XdrLargeInt {
       throw RangeError(`value too large for ${bits} bits (${this.type})`);
     }
   }
-}
 
-/**
- * Convert the raw `ScValType` string (e.g. 'scvI128', generated by the XDR) to
- * an `ScIntType` (e.g. 'i128', used by this abstraction layer).
- *
- * @param {string} scvType - the `xdr.ScValType` as a string
- *
- * @returns {ScIntType}
- */
-export function getType(scvType) {
-  return scvType.slice(3).toLowerCase();
+  static isType(type) {
+    switch (type) {
+      case 'i64':
+      case 'i128':
+      case 'i256':
+      case 'u64':
+      case 'u128':
+      case 'u256':
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Convert the raw `ScValType` string (e.g. 'scvI128', generated by the XDR)
+   * to a type description for {@link XdrLargeInt} construction (e.g. 'i128')
+   *
+   * @param {string} scvType  the `xdr.ScValType` as a string
+   * @returns {string} a suitable equivalent type to construct this object
+   */
+  static getType(scvType) {
```

Comment:
I know this is unchanged, but would it make sense to call `isType(scvType)` here and jsdoc that it throws if `scvType` invalid?

Reply:
I thought this was a good idea so I gave it a shot, but then I realized two things:

 - this shouldn't ever be called by clients, honestly, so I made it private in 7de0557 (it was only exported for tests)
 - the non-int cases are already covered by `scValToBigInt`, here: https://github.com/stellar/js-stellar-base/blob/scval-better-ints/src/numbers/index.js#L57

So this check would just add unnecessary internal overhead. If we ever need to make this public, we should definitely do the check!